### ISSUE TITLE
Resizing of the window is giving error 'A disconnected element was passed.' even after using position function.

### ISSUE THREAD

**Issue by aatif1** [anseki/leader-line#49]

**Comment by anseki** [comment]
Hi @aatif1, thank you for your report.
Could you please explain that?
For example, what did you? your browser? your code? what happened?

**Comment by aatif1** [comment]
Hi @anseki Thanks for replying.
I am using chrome and trying this:

 ` this.lines.forEach(line => {
      let start = document.getElementById(line.start);
      let end = document.getElementById(line.end);

      if (start != null && end != null) {
        line.obj = new LeaderLine(start, end, {
          color: '#002948',
          size: 2,
          path: 'straight',
          startPlug: 'disc',
          startPlugSize: 1.2,
          startSocket: 'left',
          endPlug: 'arrow3',
          endPlugSize: 1.2,
          endSocket: 'right',
        });
      }
    });`

the lines are rendering properly but whever I resize the window the lines gets all chaotic and looses connection with the divs/elements. The error is mentioned in the title. I more I resize the window the more errors I get.

**Comment by anseki** [comment]
Hmm...
I tried to resize the window, then it worked without an error.
Maybe, your code has a problem about the disconnected element (removed from document tree).
Could you reproduce the issue on https://jsfiddle.net/ ?

**Comment by aatif1** [comment]
https://github.com/aatif1/test

Maybe this will help you get the problem

**Comment by anseki** [comment]
I couldn't access that repository. It seems that that is not public repository.
Could you reproduce your issue on https://jsfiddle.net/ ?
I might be able to help you if you indicate your example code on JSFiddle.

**Comment by aatif1** [comment]
I am sorry for the inconvenience I would try jsfiddle if [stackblitz](https://stackblitz.com/edit/angular-kpsnss?file=src%2Fapp%2Febo-structure%2Febo-structure.component.html) doesn't work

**Comment by anseki** [comment]
I copied your code in [your comment](https://github.com/anseki/leader-line/issues/49#issuecomment-477641150) to JSFiddle.
https://jsfiddle.net/n5b0fud8/
Try to resize the window. This works fine without an error.
Edit this code, and could you reproduce the issue?

**Comment by anseki** [comment]
Maybe the cause of  this issue is very easy mistake, because LeaderLine throws that error only when a certain case.
Therefore the problem in your code may be found easily if you indicate your example code.

**Comment by aatif1** [comment]
Thank you for the information. The JSFiddle you provided gives a great base to understand the scenario. I will check my code once again. I am closing this issue as everything I need to know is answered by you. Thanks once again.

**Comment by anseki** [comment]
I'm glad if I can help you. :smile: 
When a disconnected element was passed, the LeaderLine throws that error.
For example, the element was removed from the document tree, or it is not added to the document tree yet.

**Comment by guy252** [comment]
> Thank you for the information. The JSFiddle you provided gives a great base to understand the scenario. I will check my code once again. I am closing this issue as everything I need to know is answered by you. Thanks once again.

hello.   did you find a solution to that issue?  im facing the same problem, using leader-line in angular9

**Comment by QbaixueQ** [comment]
> > Thank you for the information. The JSFiddle you provided gives a great base to understand the scenario. I will check my code once again. I am closing this issue as everything I need to know is answered by you. Thanks once again.
> 
> hello. did you find a solution to that issue? im facing the same problem, using leader-line in angular9

I ran into the same problem. I found that is because I draw the line repeated. you can check your code if there hide the line and draw the line again.

**Comment by sy0uta1k0n** [comment]
I met this problems, too.

I use Vue 2.0 and put leader-line init in my component mounted life cycle, but I didn't clear it in destroy (even I set this.line = null).
So 1st time, when the component mounted into my document, it worked well (no disconnected error when resize window). But when I tried to toggle my components, I found there were more than 1 leader-line DOM in my document. After that, when I resized my window, the error happened...

Here is my solution:
1. init in mounted
`mounted () {
this.line = new LeaderLine()
}`
2. destroy with leader-line api
`destroyed () {
this.line.remove()
}`
3. if you use vue 3.0, try in unmounted
